### DISCUSSION TITLE
Why would discharge capacity increase over cycle ?

### DISCUSSION THREAD

**Discussion by pkalbhor** [pybamm-team/PyBaMM#2782] [Q&A]
Disclaimer: I just started exploring PyBaMM. Please let me know if I am breaking any guidelines here in the discussion.
I tried default models like below:
```python
import pybamm
experiment = pybamm.Experiment(
    [
        ("Discharge at C/10 for 10 hours or until 3.3 V",
        "Rest for 1 hour",
        "Charge at 1 A until 4.1 V",
        "Hold at 4.1 V until 50 mA",
        "Rest for 1 hour"),
    ] * 100
)
model = pybamm.lithium_ion.SPMe()
param = pybamm.ParameterValues("Chen2020")
sim = pybamm.Simulation(model, experiment=experiment, parameter_values=param)
sol = sim.solve()
sim.plot(['Discharge capacity [A.h]'])
```
See below plot. I wonder what is logical explanation for this behavior of discharge capacity, why would it increase over time/cycle index?
![image](https://user-images.githubusercontent.com/30353865/226088413-1c1bc4a9-49ca-4cc6-9816-368b86bcced9.png)

**Comment by RuiheLi** [comment]
Hi~ This is linked to the definition of the variables "Discharge capacity [A.h]" in PyBaMM, as can be seen [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/external_circuit/base_external_circuit.py#:~:text=def%20set_initial_conditions(self,I)%20/%203600), it is just an intergration of current over time, whose integral lower limit is fixed to t=0. 
The more rigorous definition of Discharge capacity should obtained by subtacting the initial value of this varaible to the final value within a full cycle, like:
`sol.cycles[0].steps[0]["Discharge capacity [A.h]"].entries[-1]-sol.cycles[0].steps[0]["Discharge capacity [A.h]"].entries[0]`

I can see from your figure that the upper and lower boundary seem to increase equally, so the discharge capacity may always be the same.

**Reply by pkalbhor** [reply]
Thank you @RuiheLi for the reply, 
Could you please correctly point to definition of Discharge capacity? I do not see any definition in the link you have provided

**Reply by RuiheLi** [reply]
Hi @pkalbhor,

The link I provided is not the true definition of Discharge capacity recognized by the battery comminuty, it is just how PyBaMM calculate the variable Discharge capacity. You need to do what the pasted script do to get Discharge capacity.

**Reply by ch662** [reply]
Hello @RuiheLi,  I am having a similar problem. I am cycling with plating and sei formation and get a graph like this: 
<img width="446" alt="Screenshot 2023-04-04 194333" src="https://user-images.githubusercontent.com/108564982/229959685-6cb2c574-7a11-45b0-af5a-0c7a30b2d043.png">

when I am expecting something with flat minimums like this: 
<img width="124" alt="pic 2" src="https://user-images.githubusercontent.com/108564982/229959793-c2874f09-69cd-4527-957c-33c6bcd4249f.png">

This is the experiment I use: 

`C_rates = ["3C"]
#Simplification of MPCC fast-charge protocol

#want to grab steps [0, 4, 8, 12, 16, 20]
experiments2 = {}
for C_rate in C_rates:
    experiments2[C_rate] = pybamm.Experiment([
 (`
     
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at {C_rate} until 4.2V", 
     "Rest for 15 minutes",
     
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at {C_rate} until 4.2V", 
     "Rest for 15 minutes",
  
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at {C_rate} until 4.2V", 
     "Rest for 15 minutes",
     
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at {C_rate} until 4.2V", 
     "Rest for 15 minutes",
     
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at {C_rate} until 4.2V",
     "Rest for 15 minutes",
      
     f"Discharge at C/3 until 2.5V",
     "Rest for 15 minutes",
     f"Charge at C/3 until 4.2V", 
     "Rest for 15 minutes",
    )
] * 5,
termination="80% capacity"
)`

And this is the code I use to extract the discharge capacity,
`colors = ["tab:purple", "tab:cyan", "tab:red", "tab:green", "tab:blue"]
linestyles = ["dashed", "dotted", "solid"]


def plot(sims,state,cycles,num_step):
    if state == " MLA":
        color1 = "red"
    else:
        color1 = "blue"

    fig, axs = plt.subplots(1, 1, figsize=(11,7))
    for (C_rate,sim), color in zip(sims.items(),colors):
        # Isolate desired step in cycle
        cap_list=[]
        cycle_list=[]
        step_count=[]
        for i in range(cycles):                   
            for j in range(num_step):
                j=(j*4)
                if (j < num_step):
                    sol = sim.solution.cycles[i].steps[j]
                    #cap = (sol["Throughput capacity [A.h]"].entries)/0.1421 
                    #cap = (sol["Discharge capacity [A.h]"].entries)/0.1421
                    #cap = (sol.cycles[i].steps[j]["Discharge capacity [A.h]"].entries[-1]-sol.cycles[i].steps[j]["Discharge capacity [A.h]"].entries[0])/.1421
                    cap = (sol["Discharge capacity [A.h]"].entries[-1]-sol["Discharge capacity [A.h]"].entries[0])/0.1421
                    cap_f = cap #cap[len(cap)-1]
                    step_count.append(j)
                    if (cap_max > -100):
                        cap_list.append(cap_f)
                        cap_init.append(cap_i)
                        cyc = (round(j/4))+((num_step/4)*i)+1
                        cycle_list.append(cyc)

        axs.plot(cycle_list, cap_list, color=color1,linestyle=((0, (9, 10))), marker="o", label=C_rate+state)
        
        if(state=="SLA" or " SLA"):
            x=2
     
        #Put data on Graph
        axs.legend()
        axs.set_ylabel("Discharge Capacity [A.h/m2]")
        axs.set_xlabel("Cycles")

        fig.tight_layout()
    
    return fig, axs, step_count, cycle_list, cap_list`

 Using the pasted script above didn't really change anything and I was wondering if I used that wrong or if I am missing something else. 

Any help would be greatly appreciated.

**Comment by RuiheLi** [comment]
Hi~ May I ask what is the definition of `C_rates`? The first thing to do maybe to double check the solution extracted through 
```
sol = sim.solution.cycles[i].steps[j]
cap = (sol["Discharge capacity [A.h]"].entries[-1]-sol["Discharge capacity [A.h]"].entries[0])/0.1421
```
is doing extactly what you expected, by plotting the voltage curve of the specific step. 
Also, it might be helpful to edit your question again so that the whole script is clearer?
